src/buf.rs

```diff
@@ -79,4 +117,176 @@ impl BufExt for Vec<u8> {
         self.push(octal((value >> 3) & 7));
         self.push(octal(value & 7));
     }
+
+    #[inline]
+    pub(crate) fn reserve(&mut self, additional: usize) {
+        self.inner.reserve(additional)
+    }
+}
+
+impl Deref for Buf {
+    type Target = [u8];
+
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
+impl<'a> Extend<&'a u8> for Buf {
```

Comment:
Very last thing: I think we should add both `Extend<u8>` and `Extend<&'a u8>`, just for consistency with `Vec`. It's also convenient if to use with array syntax à la `.extend([..])`.